### PR TITLE
add drive archiving methods

### PR DESCRIPTION
**Pull request by andrewkrug**
* Adds methods to archive drives.

Testing plan:

* Create a mozillians group
* Add members
* Ensure they are added to the team drive.
* Remove all members and delete the group
* All members should be removed from the drive. 
* Drive will be renamed to archived_%TIMESTAMP%_%DRIVENAME%_mozilliansorg 
* The connector should block the creation of a new drive for a group with the exact same name until an admin sets a flag in DynamoDb to allow the name to be reused.

### CODE REVIEW

**Comment by andrewkrug** [comment]
@viorelaioia this is live in staging against auth0-dev and gsuite test.

**Comment by viorelaioia-zz** [comment]
I'm currently blocked on testing this, because of https://github.com/mozilla/mozillians/issues/2055

File: gsuite_driver/driver.py [hunk 1]
```diff
@@ -269,18 +293,79 @@ def all(self):
         self.drive_list = drives
         return drives
 
+    def archive(self, grouplist):
+        """Enumerate all teamDrive objects."""
+        interactive_result = ask_question(
+            interactive_mode=self.interactive_mode,
+            message='Run drive achiving routine?',
```
Comment (gdestuynder) [inline gsuite_driver/driver.py]: typo "archiving"

File: gsuite_driver/driver.py [hunk 2]
```diff
@@ -269,18 +293,79 @@ def all(self):
...
+                        interactive_mode=self.interactive_mode,
+                        message='Remove user for drive?',
+                        operation='DEL',
+                        drive_name=self.drive_name,
+                        detail=member
+                    )
+                    self.member_remove(member)
+
+                if len(self.members) == 0:
+                    new_name = 'archived_' + str(int(time.time())) + '_' + drive.get('name')
```
Comment (gdestuynder) [inline gsuite_driver/driver.py]: is a dot ok in the name? `archived_31337.31337_drivename` if yes no problem

File: gsuite_driver/settings.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 :mod:`gsuite-driver.settings` -- GSuite Driver Configuration
 * Environment variables used
 * GSUITE_DRIVER_PREFIX
+* GSUITE_DRIVER_ARCHIVING
```
Comment (gdestuynder) [inline gsuite_driver/settings.py]: due to https://github.com/andrewkrug/gsuite-community-drive-driver/blob/abf9e53aaeccf78864eec1ce45708109042f6767/gsuite_driver/main.py#L34
shouldnt this be:
`GSUITE_DRIVER_ARCHIVE` ?

File: gsuite_driver/main.py [hunk 4]
```diff
@@ -31,6 +31,30 @@ def handle(event=None, context={}):
     driver_mode = config('interactive', namespace='gsuite_driver', default='True')
     environment = config('environment', namespace='gsuite_driver', default='development')
     conformance = config('conformance', namespace='gsuite_driver', default='True')
+    archive = config('archive', namespace='gsuite_driver', default='True')
```
Comment (gdestuynder) [inline gsuite_driver/main.py]: i expected `delete = config('delete', namespace='gsuite_driver', default='True')` due to `GSUITE_DRIVER_DELETE` env var

File: README.md [hunk 5]
```diff
@@ -55,3 +55,7 @@ sls plugin install -n serverless-python-requirements
 
 sls deploy --stage dev --region us-west-2
 ```
+
+## Data Lifecycle
+
+Drives are archived when there are no members automatically.  Deleted after 180 days.
```
Comment (gdestuynder) [inline README.md:61]: it says delete after 180day and there's a `GSUITE_DRIVER_DELETE` env var, but there's no delete code

**Review by gdestuynder — CHANGES_REQUESTED** [review]
blocked due to the delete thing and the archive env var

File: gsuite_driver/driver.py (hunk 2 above)
Comment (andrewkrug) [reply gsuite_driver/driver.py]: dots are fine.

**Comment by andrewkrug** [comment]
All the requested nits have been fixed on this in addition to some fixes for the "downgraded employee" use case.  Black formatter has been run on the project.  The JSONLogger from CISv2 has been implemented for better cloudwatch logs.

**Comment by andrewkrug** [comment]
Looks like the tests are having relative path problems.  I want to move the gsuite logic into a python module like everything else.  That's a problem for future me.  Ignore travis for now since this has already landed in prod to hotfix.  Light review is fine since this code will get touched a bunch in the near future.

**Review by gdestuynder — APPROVED** [review]
>.< my eyes!

**Pull request closed without merging**